### ISSUE TITLE
Splat not supported in destructuring assingment

### ISSUE THREAD

**Issue by jaen** [google/closure-compiler#1130]
MaterialUI library has the following code:
```
let {
  selectedTime,
  mode,
  ...other,
} = this.props;
```
which is valid ES6, but trips up the compiler with the following error:
```
JSC_PARSE_ERROR: Parse error. '}' expected at line 4 character 6
      ...other,
      ^
```

**Comment by MatrixFrog** [comment]
Maybe I'm looking at the wrong part of the spec or misreading it, but it looks like the `...` is valid only in array destructuring, not object destructuring. What other ES6 tools (e.g. Traceur, Babel, browsers...) say this is valid?

**Comment by jaen** [comment]
Babel translates that without problem, traceur barfs.
As for browser Chrome beta (46.0.2490.13) does not recognise it, Firefox nightly likewise.

Now that I look at MDN's page on [destructuring assignment](https://developer.mozilla.org/pl/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment) it has a comment saying `// ES7` next to using the splat in object destructing.
It seems assuming it was ES6 syntax was an error on my part then.

I understand that this is out of scope for GClosure and I would have to put it through Babel first?

**Comment by MatrixFrog** [comment]
We will want to support ES7, but not until it's somewhat close to completion so that we know which features are almost certainly going to be in the spec and which ones aren't. So yes, this is out of scope for us, at least for the moment.
